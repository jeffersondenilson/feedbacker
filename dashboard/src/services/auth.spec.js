import mockAxios from 'axios'
import AuthService from './auth'

jest.mock('axios')

describe('AuthService', () => {
  afterEach(() => {
    jest.clearAllMocks()
  })

  it('should return a token when user login', async () => {
    const token = '123.123.123'
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: { token } })
    })

    const response = await AuthService(mockAxios)
      .login({ email: 'me@me.com', password: '123' })

    expect(response.data).toHaveProperty('token')
    expect(response).toMatchSnapshot()
  })

  it('should return an user when register', async () => {
    const user = {
      name: 'Me',
      password: '123',
      email: 'me@me.com'
    }
    mockAxios.post.mockImplementationOnce(() => {
      return Promise.resolve({ data: user })
    })

    // expect
  })
})
