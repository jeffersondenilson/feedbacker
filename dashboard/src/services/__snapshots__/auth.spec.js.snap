// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthService should return a token when user login 1`] = `
Object {
  "data": Object {
    "token": "123.123.123",
  },
  "errors": null,
}
`;

exports[`AuthService should return an user when register 1`] = `
Object {
  "data": Object {
    "email": "me@me.com",
    "name": "Me",
    "password": "123",
  },
  "errors": null,
}
`;

exports[`AuthService should return errors on login response 1`] = `
Object {
  "data": undefined,
  "errors": Object {
    "status": 404,
    "statusText": "Not Found",
  },
}
`;

exports[`AuthService should return errors on register response 1`] = `
Object {
  "data": undefined,
  "errors": Object {
    "status": 400,
    "statusText": "Bad request",
  },
}
`;
